CHANGELOG.md

```diff
@@ -1,3 +1,350 @@
+    - bash
+    - /tmp/script/run
+    volumeMounts:
+    - name: kubelet-env
+      mountPath: /etc/kubernetes/kubelet.env
+    - name: kubelet-config
+      mountPath: /etc/kubernetes/configure-kubelet-cgroup-driver
+    - name: script
+      mountPath: /tmp/script/
+  nodeName: $nodename
+  restartPolicy: Never
+  hostPID: true
+  volumes:
+  - name: kubelet-env
+    hostPath:
+      path: /etc/kubernetes/kubelet.env
+  - name: kubelet-config
+    hostPath:
+      path: /etc/kubernetes/configure-kubelet-cgroup-driver
+  - name: script
+    configMap:
+      name: script
+" | kubectl apply -f -
+
+  while true
+  do
+    complete=$(kubectl -n update-host-files get pod update-host-files-$nodename -o jsonpath='{.status.containerStatuses[-1].state.terminated.exitCode}')
+    #
+    if [ -z "${complete}" ]; then
+      echo "waiting for pod update-host-files-$nodename to finish"
+    else
+      kubectl -n update-host-files logs update-host-files-$nodename
+      break
+    fi
+    sleep 1
+  done
+
+  kubectl label node $nodename lokomotive.alpha.kinvolk.io/bgp-enabled=true
+done
+```
+
+4. Reload the new config for self-hosted kubelet
+
+  If you are running self-hosted kubelet then restart it:
+
+  ```bash
+  kubectl -n kube-system rollout restart ds kubelet
+  ```
+
+#### General upgrade steps
+
+1. Remove old asset files:
+
+  ```sh
+  rm -rf $ASSETS_DIR/cluster-assets
+  rm -rf $ASSETS_DIR/terraform-modules
+  ```
+
+2. Finally, run the following:
+
+```sh
+lokoctl cluster apply --skip-components -v
+```
+
+**NOTE:** If the upgrade process gets interrupted. Re-run `lokoctl cluster apply --skip-components -v` to proceed.
+
+The upgrade process typically takes about 10 minutes.
+After the upgrade, running `lokoctl health` should result in an output similar to the following:
+
+```sh
+Node                     Ready    Reason          Message
+
+lokomotive-controller-0  True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-0    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-1    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-2    True     KubeletReady    kubelet is posting ready status
+Name      Status    Message              Error
+
+etcd-0    True      {"health":"true"}
+```
+
+#### Upgrading native kubelets and etcd (optional)
+
+- Manually upgrade etcd following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.5.0/docs/how-to-guides/upgrade-etcd.md).
+- Manually upgrade the kubelet running on the nodes, by following the steps mentioned in the doc
+  [here](https://github.com/kinvolk/lokomotive/blob/v0.5.0/docs/how-to-guides/upgrade-bootstrap-kubelet.md).
+
+#### Updating Contour
+
+Due to some CRD changes, some manual steps are required to upgrade Contour:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/projectcontour/contour/v1.10.0/examples/contour/01-crds.yaml
+```
+
+Now it is safe to upgrade:
```

Comment:
I find this kind of tone changes confusing. I understand it's safe to upgrade, but do I *need* to upgrade? In which cases should I run this command?